js/context.js

```diff
@@ -1,252 +1,241 @@
-		} else if (!hasClass(el, className)) {
-			el.className += " " + className;
-		}
-	}
-
-	function removeClass(el, className) {
-		if (el.classList) {
-			el.classList.remove(className);
-		} else if (hasClass(el, className)) {
-			var reg = new RegExp('(\\s|^)' + className + '(\\s|$)');
-			el.className = el.className.replace(reg, ' ');
-		}
-	}
-
-
-	// mouse events
-
-	function sendClick() {
-		sendMessage({
-			action: "click"
-		});
-	}
-
-	function startDrag(e) {
-		var targetParent;
-
-		sendMessage({
-			action: "drag"
-		});
-
-		if (typeof e.dataTransfer.getData("text/html") == "undefined" || e.target.tagName == "IMG") {
-
-			targetParent = closest(e.target, "A") || document.createElement('A');
-
-			parentHTML = targetParent.cloneNode(false);
-			parentHTML.href = parentHTML.href;
-
-			targetImage = e.target.cloneNode(false);
-			targetImage.src = targetImage.src;
-
-			parentHTML.appendChild(targetImage);
-			e.dataTransfer.setData("text/html", parentHTML.outerHTML);
-		}
-
-		markletDiv.style.display = "block";
-	}
-
-	function stopDrag(e) {
-		markletDiv.style.display = "none";
-	}
-
-	function dragOver(e) {
-		e.stopPropagation();
-		e.preventDefault();
-		return false;
-	}
-
-	function dragEnter(e) {
-		sendMessage({
-			action: "dragenter"
-		});
-		return false;
-	}
-
-	function dragLeave(e) {
-		sendMessage({
-			action: "dragleave"
-		});
-		return false;
-	}
-
-	function drop(e) {
-		var data = {
-			source: location.href,
-			title: document.title
-		};
-
-		for (var i in e.dataTransfer.types) {
-			data[e.dataTransfer.types[i]] = e.dataTransfer.getData(e.dataTransfer.types[i]);
-		}
+  function createFrame(msg) {
+    markletFrame = document.createElement("iframe");
+    markletFrame.name = markletFrame.id = "arena_frame";
+    if (msg) {
+      markletFrame.src = getURL(msg);
+    } else {
+      markletFrame.src = getURL({
+        url: window.location.href,
+        title: window.document.title
+      });
+    }
+
+    document.body.appendChild(markletFrame);
+  }
+
+  function getURL(msg) {
+    baseUrl = "https://www.are.na/save/";
+
+    if(msg.options.srcUrl) {
+      path = msg.options.srcUrl;
+    } else if(msg.options.selectionText) {
+      path = msg.options.selectionText;
+    } else if(msg.options.linkUrl) {
+      path = msg.options.linkUrl;
+    } else {
+      path = msg.options.pageUrl;
+    }
+
+    url = baseUrl + encodeURIComponent(path);
+    url += "?original_source_url=" + encodeURIComponent(msg.url);
+    url += "&original_source_title=" + encodeURIComponent(msg.title);
+
+    return url;
+  }
+
+  function createTarget() {
+    markletDiv = document.createElement("div");
+    markletDiv.id = "arena_div";
+
+    document.body.appendChild(markletDiv);
+  }
+
+  function closeBookmarklet() {
+    if (markletFrame) document.body.removeChild(markletFrame);
+    if (markletDiv) document.body.removeNode(markletDiv);
+    if (markletStyle) document.body.removeNode(markletStyle);
```

Comment:
Believe we want all of these to be removeChild. Currently I get the following error:
document.body.removeNode is not a function